### PR TITLE
Feature/rkwon/auth

### PR DESCRIPTION
**Pull request by rynkwn**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -34,6 +36,20 @@ def destroy
...
+  def logged_in_user
+    unless logged_in?
+      flash[:danger] = "Please log in."
+      redirect_to login_path
+    end
+  end
+  
+  # Confirms the correct user.
+  def correct_user
+    @user = User.find(params[:id])
```
Comment (kevinchen93) [inline app/controllers/users_controller.rb]: This feels redundant. `session[:user_id]` can only hold one value and thus only map to one user. So you're either logged in or you're not, right? @rynkwn

Comment (rynkwn) [inline app/controllers/users_controller.rb]: @kevinchen93, the nuance is slightly different. I can input any id param in the url, so this function basically checks that the currently logged in user is equivalent to the id param passed in.

Comment (rynkwn) [inline app/controllers/users_controller.rb]: Ahhh. Unless you meant logged_in_user. In which case yes. I'd agree with you.

Comment (kevinchen93) [inline app/controllers/users_controller.rb]: No I was referring to `correct_user` but I take back what I said. It makes sense that you wouldn't allow me to say
```bash
curl -X DELETE '/users/<ryan-kwon-id>
```
That said, I can foresee you reusing this filter in other methods, so consider moving it to `ApplicationController`.

**Pull request merged**